04-reporting.Rmd

```diff
@@ -510,17 +509,176 @@ kableExtra: https://cran.r-project.org/web/packages/kableExtra/index.html
+
+List elements `$estimate`, `$statistic`, and `$full_result` are intended to be
+included directly into the body of your manuscript by using inline code chunks.
+For instance, you could first save the result from `apa_print()` in an output object
+
+```{r}
+out <- apa_print(
+  t.test(yield ~ N, data = npk)
+)
+```
+
+and then include a single list element by writing:
+
+```{r inline-results, eval = FALSE, highlight = FALSE}
+Plant yield indeed differed between groups, `r out$full_result`.
+```
+
+The above line is rendered as follows:
+
+> Plant yield indeed differed between groups, `r out$full_result`.
+
+
+The last element `table` is intended to be used to create results tables that may be included in the manuscript via `apa_table()`
+It will be most useful for ANOVA or linear-model results, as you will see in the next subsections.
+
+### Reporting models and tests in a table
+
+For statistical models with multiple terms (for instance, multiple regression or multi-way ANOVA) it is sometimes convenient
+to report the statistical tests of all model terms in a table.
+For this purpose, `apa_print()` returns the list element `$table` that may be included using function `apa_table()` or any other table
+(for instance, `kable()` from the **knitr** package).
+Consider the following regression model
+
+```{r}
+lm_out <- lm(formula = Post_QoL ~ Base_QoL + BDI, data = cosmetic_surgery)
+summary(lm_out)
+```
+
+To include the model table into your manuscript, first apply `apa_print()` to the model object,
+and then use `apa_table()` on the list element `$table`.
+
+```{r}
+apa_lm <- apa_print(lm_out)
+apa_table(
+  apa_lm$table
+  , caption = "A full regression table."
+)
+```
+
+
+
+
+### Fully supported analysis methods
+
+There is a plethora of possible output objects from statistical analyses in R,
+and it is of course impossible to support all of them.
+However, over the course of time, we were able to create a corpus of methods that will likely
+cover many experimental psychologists' toolboxes.
+
+- All NHST that are delivered with the base R `stats` package, which includes:
+    - $t$ for means, $\chi^2$, and nonparametric equivalents `t.test()`
+    - tests for correlation coefficients `cor.test()`
+    - (generalized) linear models
+    - ...
+    - except for pairwise comparisons (these should be done with the package **emmeans**)
+- Analysis of Variance ANOVA and ANCOVA (from packages **stats**, **car**, and **afex**)
+- Multivariate Analysis of Variance MANOVA and MANCOVA (from `stats`)
+- Hierachical (Generalized) Linear Models (from **lme4**, **lmerTest**, **nlme**, and **afex**)
+- Pairwise comparisons and planned contrasts from **emmeans**, **lsmeans**, and **glht**
+- Bayesian *t* tests and ANOVAs from **BayesFactor**
+
+There are gaps that we would like to fill, see section \@ref(future-directions).
+
+
+```{r echo = FALSE, eval = FALSE}
+library(DT)
+
+supported_functions <- list(
```

Comment:
Hmm, so while I really like listing functions I have two concerns:

1. Do all of these really work? I know I haven't tested all of these. ^^
2. Keeping this up to date is a lot of manual work. Would it be feasible to somehow create this list programmatically?

Reply:
I think all of these should work, but yes, while I started writing the list I also figured that it will be too much work to maintain the list, so I skipped it. I'm not aware of a way to list all possible functions programmatically.

Reply:
So should we than go back to just listing the supported classes, giving a few examples, and explaining how to figure out the class of an object?

Reply:
Maybe the easiest way would be to tell users to try to use `apa_print()` on their output, and it will either work or return an error together with the class of the object.

Reply:
I agree that makes sense. I still think it would be good to list the classes (for those users that can make sense of that information) and name a few basic examples, no?